Cython/Compiler/PyrexTypes.py

```diff
@@ -4404,6 +4410,110 @@ def error_condition(self, result_code):
         return "dummy"
 
 
+class PythonTypeConstructor(PyrexType):
+    """Used to help Cython interpret indexed types from the typing module (or similar)
+    """
+    is_python_type_constructor = True
+
+    def __init__(self, name, base_type=None):
+        self.name = name
+        self.base_type = base_type
+
+    def specialize_here(self, pos, env, template_values=None):
```

Comment:
We already use `specialize_here()` for fused types, but with a different signature. That's confusing.

Reply:
We use `specialize(self, value)` for `FusedTypes`. `specialize_here` was only defined for `CppClassType` (to handle templates) and has the same signature that I use here.